lib/avrora/encoder.ex

```diff
@@ -17,6 +19,37 @@ defmodule Avrora.Encoder do
     record_type: :map
   }
 
+  @doc """
+  Extract schema from binary Avro
+
+  ## Examples
+
+      ...> payload = <<0, 0, 0, 0, 8, 72, 48, 48, 48, 48, 48, 48, 48, 48, 45, 48,
+      48, 48, 48, 45, 48, 48, 48, 48, 45, 48, 48, 48, 48, 45, 48, 48, 48, 48, 48,
+      48, 48, 48, 48, 48, 48, 48, 123, 20, 174, 71, 225, 250, 47, 64>>
+      ...> Avrora.Encoder.extract_schema(payload)
+      {:ok, %Avrora.Schema{"full_name" => "io.confluent.Payment", "id" => 42}}
+  """
+  def extract_schema(payload) when is_binary(payload) do
+    case payload do
+      <<@registry_magic_bytes, <<id::size(32)>>, _body::binary>> ->
+        Resolver.resolve(id)
+
+      <<@object_container_magic_bytes, _::binary>> ->
+        {_, schema_info, _} = :avro_ocf.decode_binary(payload)
+        {
+          :ok,
+          %Avrora.Schema{
+            full_name: elem(schema_info, @object_container_full_name_pos),
+            json: OCFMapper.parse(schema_info) |> Poison.encode!
+          }
+        }
```

Comment:
What do I think, if we do have a Mapper (let's work on a name) should it be it's responsibility to extract schema from the message?

How I can imagine it:

```suggestion
      <<@object_container_magic_bytes, _::binary>> ->
        ObjectContainerFile.extract_schema(payload)
```

Reply:
I've added `Poison` because of the json string generated from the registry schema, do you know any other ways to convert a struct in a json string without it?

Reply:
I would say that the JSON field is not really needed and used just to preserve the schema. Let's focus only on 2 things then – `full_name` and `lookup_table`

Reply:
By extending `Avrora.Schema.parse`  I think we can get rid `Poison` and still have the json field for ocf take a look at this https://github.com/Strech/avrora/pull/36/files#diff-9a68d511e3a446c5046ba1963a00a33bR114

Reply:
@apellizzn let's not do the JSON, let's fill in the lookup table it can be used to encode/decode the plain message and JSON is needed only to create a lookup table. Out of lookup table you can generate the JSON easily with erlavro functionality

Reply:
Check this out https://github.com/Strech/avrora/blob/master/lib/avrora/schema.ex#L46

Reply:
Hi @Strech, I have no clue on how to fill in the lookup table using the json other than calling `Avrora.Schema.parse` 😅  Maybe it is better if I close this PR.